pkg/catalogv2/content/content.go

```diff
@@ -179,6 +186,20 @@ func (c *Manager) Icon(namespace, name, chartName, version string) (io.ReadClose
 		return nil, "", err
 	}
 
+	// Check if the repositories are managed by Rancher and if the system catalog is bundled.
+	// If so, check if the icons are available locally.
+	// If the icons are not available locally, return an error and use the default fallback logo.
+	if settings.SystemCatalog.Get() == "bundled" {
+		// If the chart icon is not an HTTP URL and the repository has a commit status,
+		// attempt to get the icon from the git repository.
+		if !isHTTP(chart.Icon) && repo.status.Commit != "" {
+			return git.Icon(namespace, name, repo.status.URL, chart)
+		}
+
+		// If the icon is not available in the git repository, use the fallback icon for airgapped environments.
+		return airgappedFallbackIcon()
```

Comment:
Also, when in airgap, why are all charts having default image instead of their own image ? We do have some images images in assets folder of rancher/charts.

Reply:
The code inside this package is fine for loading the local icon's images.
The problem is in `rancher/charts` that is not populating the `index.yaml` file with the right available icon images.

At the logos folder you provided the first image name is from `alerting-drivers.svg`:
See the `index.yaml` file, the `icon` field, it is pointing to the internet reference instead of the local repo reference which should be something like (`file://assets/logos/alerting-drivers.svg`):
```
  rancher-alerting-drivers:
  - annotations:
      catalog.cattle.io/certified: rancher
      catalog.cattle.io/display-name: Alerting Drivers
      catalog.cattle.io/kube-version: '>= 1.16.0-0 < 1.28.0-0'
      catalog.cattle.io/os: linux
      catalog.cattle.io/permits-os: linux,windows
      catalog.cattle.io/rancher-version: '>= 2.8.0-0 < 2.9.0-0'
      catalog.cattle.io/release-name: rancher-alerting-drivers
      catalog.cattle.io/type: cluster-tool
      catalog.cattle.io/upstream-version: 100.0.1
    apiVersion: v2
    appVersion: 1.16.0
    created: "2023-09-04T21:40:21.607057361+05:30"
    dependencies:
    - condition: prom2teams.enabled
      name: prom2teams
      repository: file://./charts/prom2teams
    - condition: sachet.enabled
      name: sachet
      repository: file://./charts/sachet
    description: The manager for third-party webhook receivers used in Prometheus
      Alertmanager
    digest: 392ee8c099e74e9a2b52c42d5f11cbeb158d9f79f99f66350ada13d1ad4b2d98
    icon: https://charts.rancher.io/assets/logos/alerting-drivers.svg
```

This is what I am suggesting to change by adding automating scripts in the `rancher/charts` at `Solution 2`.

Reply:
Ok, please fix this in rancher/charts repository so that Chart.Icon has the right path.